### ISSUE TITLE
Get all messages as a stream

### ISSUE THREAD

**Issue by sbellware** [message-db/message-db#27]

**Comment by shinzui** [comment]
I wasn't sure whether to comment on the PR or the issue. I'll comment here since the PR was closed.

My use case is to get all messages with a condition so I can get messages from multiple streams for a projection. There are many use cases where the read-model is across multiple stream categories, and the correct way to construct it IMO is to get messages from those categories simultaneously. Would you reconsider supporting getting all messages? EventStoreDB allows you to subscribe to the "$all" stream to solve that use case, this would offer similar functionality.

**Comment by juanpaco** [comment]
The word "projection" can be a little overloaded.  In Eventide parlance, a projection is what an autonomous component does when it receives a message it needs to handle.  It'll (some hand waving on the technical detail) get all the messages for the particular entity in question, and project them into a consistent, current-state object.

As I've seen "projection" used in the wild, it can also refer to data aggregation, or the eventually consistent construction of view data (sometimes called "read model" in DDD circles, although I think there are some nuances between "read model" and "view data").

It the former case, projecting on an `$all` stream wouldn't be appropriate because it couldn't be deterministic.  The component that owns the other streams may or may not be caught up to the present, so given when it happens to run, the result could be different.

In the latter case, I've come across a scenario where an `$all` could be useful in admin/diagnostic screens for the message store itself.  I have an open source project where I want to aggregate a list of all streams and don't want to query MessageDB for that.  So it's helpful to see every message to be able to do this.  I accomplished this by writing my own `Consumer` (http://docs.eventide-project.org/user-guide/consumers.html#example).

For normal application code, if a view data target is constructed from more than one stream category, I've accomplished this by using the stock consumers to subscribe to each category that I care about.

Don't know if these workarounds are helpful or not!

**Comment by shinzui** [comment]
> It the former case, projecting on an $all stream wouldn't be appropriate because it couldn't be deterministic. The component that owns the other streams may or may not be caught up to the present, so given when it happens to run, the result could be different.


Not necessarily. For example, if I have two aggregates, each with its own stream, and I need a read model that uses events from both streams for its projection. The first stream (S1) has events E1 and E2, and the second (S2) has events E3 and E4.

Suppose I subscribe to both streams through a single subscription (`$all` subscription with a condition clause). In that case, I can construct my view model when I encounter E4 knowing that the view is inconsistent before encountering that event. My read models are inconsistent otherwise.
